lib/interfaces/swagger-custom-options.interface.ts

```diff
@@ -10,12 +10,20 @@ export interface SwaggerCustomOptions {
   useGlobalPrefix?: boolean;
 
   /**
-   * If `false`, only API definitions (JSON and YAML) will be served (on `/{path}-json` and `/{path}-yaml`).
-   * This is particularly useful if you are already hosting a Swagger UI somewhere else and just want to serve API definitions.
+   * If `false`, the Swagger UI will not be served. Only API definitions (JSON and YAML)
+   * will be accessible (on `/{path}-json` and `/{path}-yaml`).
+   * To fully disable both the Swagger UI and API definitions, use `documentsEnabled: false`.
    * Default: `true`.
    */
   swaggerUiEnabled?: boolean;
 
+  /**
+   * If `false`, both the Swagger UI and API definitions (JSON and YAML) will be disabled.
+   * Use this option when you want to completely hide all Swagger-related endpoints.
+   * Default: `true`.
+   */
+  documentsEnabled?: boolean;
```

Comment:
```suggestion
  raw?: boolean | Array<'json' | 'yaml'>;
```
could we rename it to "raw"? raw: true means, `@nestjs/swagger` will serve raw definitions. We could also allow passing an array that determines what formats should be served, e.g. `raw: ['json']` (meaning, no yaml).